library/spdm_requester_lib/libspdm_req_encap_challenge_auth.c

```diff
@@ -101,12 +103,21 @@ libspdm_return_t libspdm_get_encap_response_challenge_auth(
     spdm_response->header.request_response_code = SPDM_CHALLENGE_AUTH;
     auth_attribute = (uint8_t)(slot_id & 0xF);
     spdm_response->header.param1 = auth_attribute;
+
     if (slot_id == 0xFF) {
         spdm_response->header.param2 = 0;
 
         slot_id = spdm_context->local_context.provisioned_slot_id;
     } else {
-        spdm_response->header.param2 = (1 << slot_id);
+        libspdm_get_cert_slot_mask_and_count(spdm_context, &slot_mask, &slot_count);
+        if (slot_count == spdm_context->local_context.slot_count) {
+            LIBSPDM_ASSERT(slot_mask);
```

Comment:
this is encap request. what if the slot is not provisioned.
Should it return some internal error?

Reply:
If we've gotten to this point in the code then the (Requester) integrator has set `CHAL_CAP=1` and `CERT_CAP=1 ^^ PUB_KEY_ID_CAP=1`, so if the integrator hasn't provisioned a certificate chain or public key then they've messed up and I don't see much point in returning an error to a buggy integrator. If they're paranoid about this they can turn on assertions.